File: src/exercises/dot_dsl/ocaml/template.ml
type attr = string * string
type node = { name : string; attrs : attr list }
type edge = { from_node : string; to_node : string; attrs : attr list }
type graph = { nodes : node list; edges : edge list; attrs : attr list }

let create_graph (items : [`Node of node | `Edge of edge | `Attr of attr] list) : graph =
  (* Your code here *)
  { nodes = []; edges = []; attrs = [] }

File: src/exercises/dot_dsl/typescript/template.ts
export class Node {
  constructor(public name: string, public attrs: Record<string, string> = {}) {}
}

export class Edge {
  constructor(public from: string, public to: string, public attrs: Record<string, string> = {}) {}
}

export class Attr {
  constructor(public key: string, public value: string) {}
}

export class Graph {
  public nodes: Node[] = [];
  public edges: Edge[] = [];
  public attrs: Record<string, string> = {};

  constructor(items: Array<Node | Edge | Attr> = []) {
    // Your code here
  }
}
